src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -535,5 +493,128 @@ private void RemoveValueFromFile<T>(string fileName, string key)
             }
         }
     }
-} // Namespace System.Management.Automation
 
+    #region GroupPolicy Configs
+
+    /// <summary>
+    /// The GroupPolicy related settings used in PowerShell are as follows in Registry:
+    ///  - Software\Policies\Microsoft\Windows\PowerShell -- { EnableScripts (0 or 1); ExecutionPolicy (string) }
+    ///      SubKeys                  Name-Value-Pairs
+    ///      - ScriptBlockLogging     { EnableScriptBlockLogging (0 or 1); EnableScriptBlockInvocationLogging (0 or 1) }
+    ///      - ModuleLogging          { EnableModuleLogging (0 or 1); ModuleNames (string[]) }
+    ///      - Transcription          { EnableTranscripting (0 or 1); OutputDirectory (string); EnableInvocationHeader (0 or 1) }
+    ///      - UpdatableHelp          { DefaultSourcePath (string) }
+    ///      - ConsoleSessionConfiguration { EnableConsoleSessionConfiguration (0 or 1); ConsoleSessionConfigurationName (string) }
+    ///  - Software\Policies\Microsoft\Windows\EventLog
+    ///     SubKeys                   Name-Value-Pairs
+    ///      - ProtectedEventLogging  { EnableProtectedEventLogging (0 or 1); EncryptionCertificate (string[]) }
+    ///
+    /// The JSON representation is in sync with the 'PowerShellPolicies' type. Here is an example:
+    /// {
+    ///   "PowerShellPolicies": {
+    ///     "ScriptExecution": {
+    ///       "ExecutionPolicy": "RemoteSigned"
+    ///     },
+    ///     "ScriptBlockLogging": {
+    ///       "EnableScriptBlockInvocationLogging": true,
+    ///       "EnableScriptBlockLogging": false
+    ///     },
+    ///     "ProtectedEventLogging": {
+    ///       "EnableProtectedEventLogging": false,
+    ///       "EncryptionCertificate": [
+    ///         "Joe"
+    ///       ]
+    ///     },
+    ///     "Transcription": {
+    ///       "EnableTranscripting": true,
+    ///       "EnableInvocationHeader": true,
+    ///       "OutputDirectory": "c:\tmp"
+    ///     },
+    ///     "UpdatableHelp": {
+    ///       "DefaultSourcePath": "f:\temp"
+    ///     },
+    ///     "ConsoleSessionConfiguration": {
+    ///       "EnableConsoleSessionConfiguration": true,
+    ///       "ConsoleSessionConfigurationName": "name"
+    ///     }
+    ///   }
+    /// }
+    /// </summary>
+    internal sealed class PowerShellPolicies
+    {
+        public ScriptExecution ScriptExecution { get; set; }
+        public ScriptBlockLogging ScriptBlockLogging { get; set; }
+        public ModuleLogging ModuleLogging { get; set; }
+        public ProtectedEventLogging ProtectedEventLogging { get; set; }
+        public Transcription Transcription { get; set; }
+        public UpdatableHelp UpdatableHelp { get; set; }
+        public ConsoleSessionConfiguration ConsoleSessionConfiguration { get; set; }
+    }
+
+    internal abstract class PolicyBase { }
+
+    /// <summary>
+    /// Setting about ScriptExecution
+    /// </summary>
+    internal sealed class ScriptExecution : PolicyBase
```

Comment:
Maybe `ScriptExecutionPolicy`?

Reply:
This policy controls script execution. In `Local Group Policy Editor`, its name is "Turn on Script Execution".

Reply:
I think I misunderstood your comment :) You mean it may be more readable if we add 'Policy' as a suffix to those type names. I would like to keep those names as close to GP registry key as possible, especially since we are still using GP in PSCore.

Reply:
Yes, I meant that it may be more readable if we add 'Policy' as a suffix to those type names. It is not blocking comment.